File: vibe-coding/todo-app/backend_web/src/modules/lists_mod/application/delete_list/delete_list_dto.py
from dataclasses import dataclass
from typing import Any, Self, final

from src.modules.lists_mod.domain.enums.list_field_enum import ListFieldEnum


@final
@dataclass(frozen=True, slots=True)
class DeleteListDto:
    """Entrada del caso de uso DeleteList."""

    list_id: int

    @classmethod
    def from_primitives(cls, primitives: dict[str, Any]) -> Self:
        return cls(list_id=int(primitives.get(ListFieldEnum.LIST_ID, 0) or 0))

File: vibe-coding/todo-app/frontend_web/src/modules/lists_mod/application/delete_list/DeleteListDto.ts
/** Entrada del caso de uso DeleteList. */
export class DeleteListDto {
    private constructor(public readonly listId: number) {}

    public static fromPrimitives(primitives: Record<string, unknown>): DeleteListDto {
        return new DeleteListDto(Number(primitives["listId"] ?? 0));
    }
}
